cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeElasticsearchSinkTest.java

```diff
@@ -0,0 +1,166 @@
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.node.NodeBuilder.nodeBuilder;
+
+/**
+ * Test for {@link RealtimeElasticsearchSink}
+ */
+public class RealtimeElasticsearchSinkTest extends TestBase {
+  private static final Id.Namespace NAMESPACE = Id.Namespace.DEFAULT;
+  private static final Id.ApplicationTemplate TEMPLATE_ID = Id.ApplicationTemplate.from("ETLRealtime");
+  private static final Gson GSON = new Gson();
+
+  private Client client;
+  private Node node;
+
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void beforeTest() throws Exception {
+    ImmutableSettings.Builder elasticsearchSettings = ImmutableSettings.settingsBuilder()
+      .put("path.data", tmpFolder.newFolder("data"))
+      .put("cluster.name", "testcluster");
+    node = nodeBuilder().settings(elasticsearchSettings.build()).client(false).node();
+    client = node.client();
+  }
+
+  @After
+  public void afterTest() {
+    node.close();
+  }
+
+  @BeforeClass
+  public static void setupTests() throws IOException {
+    // TODO: should only deploy test source and Elasticsearch sink (JIRA CDAP-3322)
+    addTemplatePlugins(TEMPLATE_ID, "realtime-sources-1.0.0.jar",
+                       DataGeneratorSource.class, JmsSource.class, KafkaSource.class,
+                       TwitterSource.class, SqsSource.class);
+    addTemplatePlugins(TEMPLATE_ID, "realtime-sinks-1.0.0.jar",
+                       RealtimeCubeSink.class, RealtimeTableSink.class,
+                       StreamSink.class, RealtimeElasticsearchSink.class);
+    addTemplatePlugins(TEMPLATE_ID, "transforms-1.0.0.jar",
+                       ProjectionTransform.class, ScriptFilterTransform.class,
+                       StructuredRecordToGenericRecordTransform.class);
+    deployTemplate(NAMESPACE, TEMPLATE_ID, ETLRealtimeTemplate.class,
+                   PipelineConfigurable.class.getPackage().getName(),
+                   ETLStage.class.getPackage().getName(),
+                   RealtimeSource.class.getPackage().getName());
+  }
+
+  @Test
+  @Category(SlowTests.class)
+  public void testESSink() throws Exception {
+    ETLStage source = new ETLStage("DataGenerator", ImmutableMap.of(DataGeneratorSource.PROPERTY_TYPE,
+                                                                    DataGeneratorSource.TABLE_TYPE));
+    try {
+      ETLStage sink = new ETLStage("Elasticsearch",
+                                   ImmutableMap.of(Properties.Elasticsearch.TRANSPORT_ADDRESSES,
+                                                   InetAddress.getLocalHost().getHostName() + ":9300",
```

Comment:
So we are expecting the port to be always 9300? Will it cause a problem if that port is not available?